### PR TITLE
Document how to Update Data Builder Docs

### PR DESCRIPTION
**Pull request by ghickman**
This adds a section on how to update parts of documentation which come from Data Builder.  It assumes the pipeline itself does not need updating, and that we'll document that elsewhere.  My intention was to mirror the old cohortextractor flow of "I have changed something in cohortextractor, how do I update the docs page?"

@StevenMaude – re that elsehwere, do you think that should be another section on this page or live in the relevant projects?

### CODE REVIEW

File: docs/updating-the-docs.md
Comment (iaindillingham) [inline docs/updating-the-docs.md:51]: Nitpick: Titles in this section seem not to capitalize the first letter of each word. _Data Builder_ is an exception, because it's a proper noun (cf. common noun). By this reasoning, I think that _Backend_ and _Contract_ should be _backend_ and _contract_.

File: docs/updating-the-docs.md
Comment (iaindillingham) [inline docs/updating-the-docs.md:63]: Same nitpick as before 🙂

File: docs/updating-the-docs.md
Comment (iaindillingham) [inline docs/updating-the-docs.md:59]: We refer to this as "the Data Builder repo" below. Could we do the same here? It indicates what "Data Builder" means in this context; that is, the repository.

File: docs/updating-the-docs.md
Comment (iaindillingham) [inline docs/updating-the-docs.md:66]: I think we should emphasize that this is done after a new version of databuilder is released. (Do all merges to `main` prompt a new release?)

**Review by iaindillingham — APPROVED** [review]
Nitpicks ⚠️ I'll approve, but @StevenMaude should also 👀.

File: docs/updating-the-docs.md
Comment (ghickman) [reply docs/updating-the-docs.md:59]: I've not mentioned the repo here because it's not the repo per se, but the tool itself that is the component.  I linked to the repos for each component because that's where the one can make changes.  I don't feel strongly about it though, do you think it's worth changing still?

File: docs/updating-the-docs.md
Comment (ghickman) [reply docs/updating-the-docs.md:66]: > Do all merges to main prompt a new release?

Yep!  The `build_and_deploy` workflow tags the repo based on the semantic commit used, and builds a new docker image.  Technically there's a race condition where you could hit merge, and then run `just databuilder/upgrade-databuilder` before the tagging flow finishes, but I suspect it's very unlikely to happen and I'm not sure it would be a valuable addition to the docs?

File: docs/updating-the-docs.md
Comment (iaindillingham) [reply docs/updating-the-docs.md:59]: That's reasonable. At some point, it would be useful to note down what we're referring to with Data Builder, databuilder, etc. I don't feel strongly about it, though 🙂

Comment (ghickman) [reply docs/updating-the-docs.md:59]: The [canonical name is `Data Builder`](https://app.shortcut.com/ebm-datalab/epic/372/rename-to-data-builder?vc_group_by=day&ct_workflow=all&cf_workflow=500000005), but I suspect many of us are referring to `databuilder` accidentally or to mean the repo without thinking about it!

File: docs/updating-the-docs.md
Comment (iaindillingham) [reply docs/updating-the-docs.md:66]: I think I've misunderstood. The `main` workflow's `tag-new-version` step doesn't always create a new release (only for `fix`, `perf`, `feat`, and `BREAKING CHANGE`). If I merged a `docs` change to `main`, such as changing the docstring for `PatientDemographics`, then a new release wouldn't be created, so this step would be a noop, wouldn't it?

Comment (ghickman) [reply docs/updating-the-docs.md:66]: TIL `docs` doesn't create a new release… arg.

I'll amend to make this part conditional, thanks for catching this!

Comment (iaindillingham) [reply docs/updating-the-docs.md:66]: Thank you 🙂

Comment (iaindillingham) [reply docs/updating-the-docs.md:66]: `github-tag-action` never fails to be confusing.

Comment (ghickman) [reply docs/updating-the-docs.md:66]: Oh wait, we actually need a new version of databuilder to be built so we can use the tooling here because we only install a versioned build of it 😩

The alternative is to run the command with a local copy of the databuilder and then copy `public_docs.json` over, but I think that's going to make the documentation here very complicated.  However, the alternative feels like a foot gun waiting to happen 🤔

File: docs/updating-the-docs.md
Comment (StevenMaude) [inline docs/updating-the-docs.md:53]: suggestion:

* Maybe clarify briefly why you need both (linking to #493 might suffice)
* Maybe worth warning that this isn't supported for Windows as yet, if that's the case still. [^1]

[^1]: We could try Gitpod, but not sure how this would work out.

File: docs/updating-the-docs.md
Comment (StevenMaude) [inline docs/updating-the-docs.md:55]: thought: I suspect there's more effort on the reader's part when trying to work out whether "Data Builder" and "databuilder" are the same or different here. My opinion rather than fact, though.

I think I'd always favour "Data Builder" unless explicitly referring to actually code use — that is, `import databuilder`. For the case of the repository, I'd probably write "Data Builder repository" and link to the repository. Anyone reading, feel free to suggest counter-examples.

(We may have inconsistencies elsewhere and we can tidy these up, before we make the documentation more public.)

File: docs/updating-the-docs.md
Comment (StevenMaude) [reply docs/updating-the-docs.md:59]: thought: I suspect there's more effort on the reader's part when trying to work out whether "Data Builder" and "databuilder" are the same or different here. My opinion rather than fact, though.

I think I'd always favour "Data Builder" unless explicitly referring to actually code use — that is, `import databuilder`. For the case of the repository, I'd probably write "Data Builder repository" and link to the repository. Anyone reading, feel free to suggest counter-examples.

(We may have inconsistencies elsewhere and we can tidy these up, before we make the documentation more public.)

File: docs/updating-the-docs.md
Comment (StevenMaude) [inline docs/updating-the-docs.md:55]:
```suggestion
Some of the Data Builder information in this OpenSAFELY documentation is generated from the [Data Builder source code](https://github.com/opensafely-core/databuilder).
```

File: docs/updating-the-docs.md
Comment (StevenMaude) [reply docs/updating-the-docs.md:66]: comment: I think this still need addressing :arrow_up:

Comment (StevenMaude) [reply docs/updating-the-docs.md:66]: suggestion: The quick fix might just be to say:
* a current limitation is that you need to wait for a new release before rebuilding the generated content
* note that releases are only made for certain pull requests

**Comment by StevenMaude** [comment]
summary: This looks mostly fine as is. I think it'll be subject to future refinement, in any case.

File: docs/updating-the-docs.md
Comment (StevenMaude) [reply docs/updating-the-docs.md:53]: The Justfile seems bash-heavy. My assumption is that trying to run the same commands in Windows will fail.

**Comment by StevenMaude** [comment]
> It assumes the pipeline itself does not need updating, and that we'll document that elsewhere.
> …
> re that elsehwere, do you think that should be another section on this page or live in the relevant projects?

Sorry @ghickman: I entirely skimmed this question on first look here :face_with_spiral_eyes:

To answer, my instinct is that such "pipeline" information could go further down this same page, if it might encompass *multiple different projects* where the outputs are in this repository. Perhaps with a warning that the guidance is more for software maintainers than researchers.

It's perhaps easier to see when there's a concrete example of what to add. In any case, I think it's fine not to bother considering that for this PR.

**Comment by ghickman** [comment]
@StevenMaude – updated per feedback, could you take another look over it?

File: docs/updating-the-docs.md
Comment (StevenMaude) [inline docs/updating-the-docs.md:55]:
```suggestion
!!! note "This workflow is not supported, and has not been tested, on Windows."
```

**Comment by StevenMaude** [comment]
@ghickman: I added one more small suggestion to consider. Otherwise, this looks good to merge.

**Review by StevenMaude — APPROVED** [review]

**Pull request merged**
